File: 219/cg_219_easy.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """

        # Fast solution based on sliding window. Set s contains
        # elements currently inside the window.
        s = set()
        for i in range(0, len(nums)):
          if i > k:
            s.remove(nums[i - k - 1])

          if nums[i] in s:
            return True

          s.add(nums[i])

        return False

File: 219/cg_219_easy.java
public class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, List<Integer>> m = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            if (!m.containsKey(nums[i])) {
                m.put(nums[i], new ArrayList<Integer>());
            }
            
            List<Integer> indices = m.get(nums[i]);
            
            indices.add(i);
            
            if (indices.size() > 1 && indices.get(indices.size() - 1) - indices.get(indices.size() - 2) <= k) {
                return true;
            }
        }
        
        return false;
    }
}
